src/c++/perf_analyzer/inference_profiler.cc

```diff
@@ -1588,8 +1605,27 @@ InferenceProfiler::SummarizeServerStatsHelper(
 
   const auto& end_itr = end_status.find(this_id);
   if (end_itr == end_status.end()) {
-    return cb::Error(
-        "missing statistics for requested model", pa::GENERIC_ERROR);
+    if (parser_ == nullptr || !parser_->TopLevelRequestCachingEnabled()) {
+      return cb::Error(
+          "missing statistics for requested model", pa::GENERIC_ERROR);
+    } else {
+      server_stats->inference_count = 0;
```

Comment:
Can you add a comment explaining why all this initialization is needed here?

Reply:
And does it make sense to add a `ServerSideStats::Reset()` or something that does all this instead?

Reply:
I had a conflicting idea about this. The perf_analyzer checks if every model has stats.
In case of a cache_hit, the composing models do not load and they don't have any stats.
I wanted to leave the stats empty but felt it might be confusing if someone interprets it as an error scenario.

Reply:
At a minimum let's add a comment explaining it, and consider a helper reset function if that correctly captures what we're doing here

Reply:
Ok, I will do that.